### ISSUE TITLE
Bracket highlighting with word boundaries do not work with periods

### ISSUE THREAD

**Issue by PsychoNineSix** [microsoft/vscode#152429]
Issue Type: <b>Bug</b>

In issue #132162 word boundaries were fixed for most cases, but not for records/namespaces.

```pascal
Point := PEnd; // works
Line.End; // period at the start works
Point.X := PEnd.X; // period at the end breaks bracket matching
```

VS Code version: Code 1.68.1 (30d9c6cd9483b2cc586687151bcbcd635f373630, 2022-06-14T12:48:58.283Z)
OS version: Windows_NT x64 10.0.19044
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz (12 x 2592)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.81GB (4.28GB free)|
|Process Argv|--crash-reporter-id f5c1b75d-5924-472b-bb20-95114cd3aa84|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (19)</summary>

Extension|Author (truncated)|Version
---|---|---
monokai-charcoal-high-contrast|74t|3.4.0
markdown-mermaid|bie|1.14.0
mermaid-markdown-syntax-highlighting|bpr|1.2.4
gitlens|eam|12.1.0
vs-code-xml-format|fab|0.1.5
gitlab-workflow|Git|3.47.1
path-autocomplete|ion|1.19.1
vscode-peacock|joh|4.0.1
docomment|k--|0.1.31
hex-hover-converter|maz|1.2.1
csharp|ms-|1.24.4
azure-account|ms-|0.10.1
vsliveshare|ms-|1.0.5615
vscode-yaml|red|1.8.0
vsc-workspace-sidebar|ske|1.4.1
errorlens|use|3.5.1
vscode-icons|vsc|11.12.0
omnipascal|Wos|0.19.0
markdown-all-in-one|yzh|3.4.3

</details>

**Comment by PsychoNineSix** [comment]
Just noticed this would also be fixed with #150440

**Comment by PsychoNineSix** [comment]
> Just noticed this could also be fixed with #150440

Not the case, the definition of the bracket is a complete word, which is not fixed by the
